models/models.js

```diff
@@ -7,30 +7,64 @@ exports.selectTopics = () => {
 };
 
 exports.selectArticles = () => {
-return db
-  .query(
-    `
+  return db
+    .query(
+      `
   SELECT articles.*, 
   COUNT(comment_id) AS comment_count FROM articles
   LEFT JOIN comments ON articles.article_id = comments.article_id
   GROUP BY articles.article_id
-  ORDER BY created_at DESC;`)
-  .then((result) => {
+  ORDER BY created_at DESC;`
+    )
+    .then((result) => {
       return result.rows;
-    })
-}
+    });
+};
 
 exports.selectArticlesByID = (article_id) => {
-  const queryText = `SELECT * FROM articles WHERE article_id=$1`;
+  const queryText = `
+  SELECT * FROM articles 
+  WHERE article_id=$1`;
   const queryVals = [article_id];
   return db.query(queryText, queryVals).then((article) => {
     if (article.rows.length === 0) {
       return Promise.reject({
         status: 404,
-        msg: "Bad request"
+        msg: "Bad request",
       });
     } else {
       return article.rows[0];
     }
-    });
+  });
+};
+
+exports.selectCommentsByArticleID = (article_id) => {
+  const queryStr = `
+  SELECT * FROM comments 
+  WHERE article_id = $1
+  ORDER BY created_at DESC;`;
+
+  return db.query(queryStr, [article_id]).then((articleComments) => {
+    if (articleComments.rows.length === 0) {
+      return Promise.reject({
+        status: 404,
+        msg: "Bad request",
```

Comment:
The issue here with throwing a `404` when the comments array is empty - is that some articles have no comments and that is okay, we would still expect to return a `200` as long as the `article_id` is valid and in the database

Reply:
to get around this you can use your previously built model `selectArticleById` together with this model - in the controller for gettingComments call both models in a `Promise.all` - if the `selectArticleByID` is given an `article_id` that is not in the database it will throw an error to the `.catch` - if it does not throw an error we can safely assume it was found in the database and the comments array can be sent back to the client regardless of whether or not it is empty